plugin/src/main/java/org/opensearch/ml/settings/MLCommonsSettings.java

```diff
@@ -43,4 +43,12 @@ private MLCommonsSettings() {}
 
     public static final Setting<Integer> ML_COMMONS_MAX_UPLOAD_TASKS_PER_NODE = Setting
         .intSetting("plugins.ml_commons.max_upload_model_tasks_per_node", 10, 0, 10, Setting.Property.NodeScope, Setting.Property.Dynamic);
+
+    public static final Setting<String> ML_COMMONS_TRUSTED_URL_REGEX = Setting
+        .simpleString(
+            "plugins.ml_commons.trusted_url_regex",
+            "^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]",
```

Comment:
does this regex cover http & https both?

Reply:
Yes, I tested manually , I will add more test cases in next PR.

Reply:
Yes, the question mark means 0 - 1.